test/ComponentInstallerTest.php

```diff
@@ -191,20 +195,40 @@ private function createOutputAssertions(array $informations): void
         $consecutiveArguments = [];
 
         foreach ($informations as $information) {
-            $consecutiveArguments[] = [
-                self::callback(static function (string $argument) use ($information): bool {
-                    return preg_match(
-                        sprintf('/%s/', preg_quote($argument, '/')),
-                        $information
-                    ) !== false;
-                }),
-            ];
+            $consecutiveArguments[] = self::callback(static function (string $argument) use ($information): bool {
+                return preg_match(
+                    sprintf('/%s/', preg_quote($argument, '/')),
+                    $information
+                ) !== false;
+            });
         }
 
+        $consecutive = new class ($consecutiveArguments) extends Constraint {
+            /**
+             * @param list<Constraint> $assertions
+             */
+            public function __construct(private array $assertions)
+            {
+                $this->assertions[] = new IsAnything();
+            }
+
+            public function matches(mixed $other): bool
+            {
+                $assertion = current($this->assertions);
+                next($this->assertions);
+                return $assertion->matches($other);
+            }
+
+            public function toString(): string
+            {
+                return current($this->assertions)->toString();
+            }
+        };
+
         $this->io
             ->expects(self::exactly(count($consecutiveArguments)))
             ->method('write')
-            ->withConsecutive(...$consecutiveArguments);
+            ->with($consecutive);
```

Comment:
I think this approach is fine: we're recording stuff, then verifying order at the end.

Reply:
This one essentially a custom constraint composing actual constraints and iterating over them on consecutive invocations. If phpunit decides to call constraint multiple times in the future for whatever reason this will break but it is fairly unlikely.

Unlike `withConsecutive()` this constraint can match only one parameter but here that is exactly what is needed.

I took this approach over `willReturnCallback()` since I want mock object to record assertion failures as it normally does instead of risking them getting caught by try-catch somewhere in the tested call stack.